packages/artillery/lib/cmds/run.js

```diff
@@ -529,6 +536,20 @@ async function sendTelemetry(script, flags, extraProps) {
   }
 }
 
+function checkDirExists(output) {
+  if (!output) {
+    return;
+  }
+  // If destination is a file check only path to its directory
+  const exists = path.extname(output)
```

Comment:
Relying on `extname()` to check whether a path points to a file or a folder is not ideal as it will break on directories with `.` characters in them. E.g. if the user passes `-o my.reports/` to Artillery, the result of `path.dirname()` will be "." rather than `my.reports`.

[`fs.lstat().isDirectory()`](https://nodejs.org/api/fs.html#fslstatpath-options-callback) would be a more reliable way to check whether a path points to a directory or a file. Once we know which one it is, we can check if the directory exists.

Reply:
Thank you for the review and suggestions @hassy !  I see now how using path.extname() could be problematic.

The reason I didn't use `fs.lstat().isDirectory()` or `fs.statSync().isDirectory` (which is used in the original getLogFileName function) in the first place, is because these functions return the `fs.Stats` object (and therefore the `isDirectory()` method) only if the path provided actually exists. If the path does not exist  error is thrown (or returned in case of `fs.lstat()` ).
Due to this unfortunately I could not find a way to use them to find out whether a path is a file or a directory prior to checking its existence. 
But I am new at this and could be missing something..

We could maybe do something like this:

```
function checkDirExists(output) {

  // We check first if parent directory path is valid
  const parent = fs.existsSync(path.dirname(output));
  const noDirError = new Error(`Directory ${output} does not exist.`);

  if (!parent) {
    throw noDirError;
  }

  // If parent is valid we check if the whole path is valid
  // If the path does not exist and the last character of path is a path separator
  // then it is considered a nonexistent directory
  if (!fs.existsSync(output) && output.at(-1) === (path.sep || '/')) {
    throw noDirError;
  }
}
```
I believe this might screen the output string the proper way. What do you think of something similar to above?